Assets/Scripts/Utility/WeaponBasics.cs

```diff
@@ -202,6 +203,130 @@ public static string GetMagicAnimFilename(ElementTypes elementType)
             }
         }
 
+        public static string GetModdedWeaponFilename(DaggerfallUnityItem weapon)
+        {
+            if (weapon.IsEnchanted)
+            {
+                if (weapon.IsArtifact)
+                {
+                    switch (weapon.TemplateIndex)
+                    {
+                        case (int)Weapons.Dagger:
+                            return "MEHRUNESRAZOR.CIF";
+                        case (int)Weapons.Staff:
+                            if (weapon.ItemName == "The Wabbajack")
```

Comment:
Best to avoid a string literal check here, as someone might translate the name in future, e.g. "Der Wabbajack" or "
Le Wabbajack" or even "ワバジャック" :)

There must be some other unique way of identifying Wabbajack item here.

Reply:
Yeah, I did not consider that fact, I'll try and use one of the other ways that it is done in the code. Which I think is mainly through checking unique spell effects for some of them. The reason I mainly went with the string check being I was not really 100% understanding how those unique effects checks were working. Also a note that there is at least a few examples in "FormulaHelper.cs" that is checking artifacts based on their literal item-name.

Reply:
I was looking at some possible alternative in the code, but I'm sort of concerned that the only good potential alternative I found might be slower? That being checking the "SpecialArtifactEffects", but at least in this example it seems to need to loop through the entire item.LegacyEnchantments to get these values? I might just be misunderstanding it though.

![Capture](https://user-images.githubusercontent.com/39843717/180582377-c409410a-b8e9-4e43-8868-50b2e750d14a.PNG)

This is also where I got the idea of using the literal item-name as a way of identification, since it's used at the bottom there for Chrysamere and Staff of Magnus, which apparently don't have any "unique" effect like some of the other artifacts.

Reply:
> Also a note that there is at least a few examples in "FormulaHelper.cs" that is checking artifacts based on their literal item-name

Right you are. :) For my part, I'm gradually (slowly) working back through all the use of string literals and trying to migrate these over to support localization in some way. This has been a real agony for me and I often feel like I'm barely making progress. What I'm trying to do (if possible) is not introduce any more cases of this in new code. That way I only have to fix the old code and not keep chasing new instances being added by new code. The gap here of course is that handling for this hasn't been added yet so there's no good solution for you to move to either.

I'm sorry if that's adding complexity for you. I can definitely make an exception in this case if we can't find another way to identify cleanly. What's one more. :)

Reply:
Yeah, it does not help obviously that artifacts are such a "unique" case. I'm sure this sort of thing happens in other code-bases within the Elder Scrolls series, just sort of hard-coded values and names. I'll see if I can do something with the unique effect checking like is shown in that screenshot, but I don't have too much confidence if it will be much better from me, lol.

Reply:
It's fine, let's not stress this one for now. :) I'll review this one better in another pass when I have a bit more time and will raise any better ideas if one comes up.

Reply:
I just made another commit that will hopefully be a less string literal solution, using the same principle as the attached screenshot, just slightly modified.